cases/serializers.py

```
@@ -301,6 +302,63 @@ class Meta(CaseSerializer.Meta):
+    def to_internal_value(self, data):
+        from django.core.exceptions import ValidationError as DjangoValidationError
+        from django.core.validators import URLValidator
+
+        validator = URLValidator()
+
+        if isinstance(data, str):
+            stripped = data.strip()
+            try:
+                validator(stripped)
+            except DjangoValidationError:
+                raise serializers.ValidationError("Enter a valid URL.")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)